torchvision/transforms/v2/_auto_augment.py

```diff
@@ -82,10 +127,23 @@ def _apply_image_or_video_transform(
         interpolation: Union[InterpolationMode, int],
         fill: Dict[Type, datapoints._FillTypeJIT],
     ) -> Union[datapoints._ImageType, datapoints._VideoType]:
-        fill_ = fill[type(image)]
+        return self._apply_transform(
+            image, transform_id=transform_id, magnitude=magnitude, interpolation=interpolation, fill=fill
+        )
+
+    def _apply_transform(
+        self,
+        inpt: datapoints._InputType,
+        transform_id: str,
+        magnitude: float,
+        interpolation: InterpolationMode,
+        fill: Dict[Type, datapoints._FillTypeJIT],
+    ) -> Union[datapoints._ImageType, datapoints._VideoType]:
+        # TODO: magnitude is not always float, it could be int (e.g., posterize).
```

Comment:
In `_AUGMENTATION_SPACE`, they are already converted to `int` format, so we may need to change magnitude type annotation to `Union[float, int]`